lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -101,31 +101,30 @@ class EngineAutofillForm {
     // If there is only one text field in the autofill model, `fields` will be
     // null. `focusedElementAutofill` contains the information about the one
     // text field.
-    final bool singleElement = (fields == null);
-    final AutofillInfo focusedElement =
-        AutofillInfo.fromFrameworkMessage(focusedElementAutofill);
-    final Map<String, html.HtmlElement> elements = <String, html.HtmlElement>{};
-    final Map<String, AutofillInfo> items = <String, AutofillInfo>{};
+    final Map<String?, html.HtmlElement> elements = <String?, html.HtmlElement>{};
+    final Map<String?, AutofillInfo> items = <String?, AutofillInfo>{};
     final html.FormElement formElement = html.FormElement();
 
     // Validation is in the framework side.
     formElement.noValidate = true;
 
     _hideAutofillElements(formElement);
 
-    if (!singleElement) {
-      for (Map<String, dynamic> field in fields) {
+    if (fields != null) {
```

Comment:
why did we removed singleElement? It was making the code easy to understand.

Reply:
Yeah, this is unfortunate. The type inference is unable to infer non-null via a boolean local. I filed an issue for the Dart SDK: https://github.com/dart-lang/sdk/issues/42469.